### PR TITLE
Global e2e tests refactoring and improvement

### PR DESCRIPTION
**Pull request by avozicov**
- Implemented POMs for:
   - api-page
   - contact-page
   - demo-modal
   - footer
   - header
   - home-page
   - quick-links
   - social-links
- Refactored all tests scripts to use page objects described in "pages" directory 
- Added new script for "Social Links" validation
- Added new script for "API page" content assert 
- Fixed const names to provide more context of selectors and actions

### CODE REVIEW

File: test/e2e/tests/assertApiPage.e2e.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { Selector } from 'testcafe';
+import ApiPage from '../pages/api-page.po';
+
+const apiPage = new ApiPage();
+
+fixture`Check valid content is displayed on "API" page`
+  .page`https://www-stage.adtechmedia.io/api/`;
```
Comment (AlexanderC) [inline test/e2e/tests/assertApiPage.e2e.js]: use base host (e.g. `https://www-stage.adtechmedia.io`) from a `config.js` like this: `config.www_base_host`

```javascript
// config.js

export default {
  www_base_host: process.env.WWW_BASE_HOST || 'https://www-test.adtechmedia.io',
}
```

You can run tests like `WWW_BASE_HOST='https://www-stage.adtechmedia.io' recink run e2e`

File: test/e2e/tests/assertFooter.e2e.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import FooterLinks from '../pages/footer.po';
+
+const footerLinks = new FooterLinks();
+
+fixture`Check valid content and links are displayed on website footer`
+  .page`https://www-stage.adtechmedia.io/`;
+
+test('Check "Mitoc Group" link is displayed on the page footer', async t => {
+  await t
+    .resizeWindow(1920, 1080)
```
Comment (AlexanderC) [inline test/e2e/tests/assertFooter.e2e.js]: Try setting default window size from config (see nighmare docs)

File: test/e2e/tests/submitDemoRequest.e2e.js [hunk 3]
```diff
@@ -1,32 +1,32 @@
...
     }).visible).ok();
 
-  const requestDemoPage = new RequestDemoPage();
+  const requestDemoForm = new RequestDemoForm();
 
   await t
-    .typeText(requestDemoPage.demoFormEmailField, 'sakusha@yahoo.ca')
-    .typeText(requestDemoPage.demoFormNamelField, 'Test User')
-    .click(requestDemoPage.demoFormSubmitButton, { speed: 0.5 });
+    .typeText(requestDemoForm.demoFormEmailField, 'hellfire@yahoo.com')
```
Comment (AlexanderC) [inline test/e2e/tests/submitDemoRequest.e2e.js]: Please use http://chancejs.com library for generating random test data

File: test/e2e/tests/submitDemoRequest.e2e.js [hunk 4]
```diff
@@ -1,32 +1,32 @@
...
+    .click(requestDemoForm.demoFormSubmitButton, { speed: 0.5 });
 
   await t
-    .expect(Selector(requestDemoPage.demoFormResponseText).with({
+    .expect(Selector(requestDemoForm.demoFormResponseText).with({
       selectorTimeout: 5000,
       visibilityCheck: true,
     }).visible).ok()
-    .expect(Selector(requestDemoPage.demoFormResponseText).innerText).eql('\nThank you for the interest in AdTechMedia WordPress Plugin\nPlease check your inbox to schedule your demo.\n');
+    .expect(Selector(requestDemoForm.demoFormResponseText).innerText).eql('\nThank you for the interest in AdTechMedia WordPress Plugin\nPlease check your inbox to schedule your demo.\n');
```
Comment (AlexanderC) [inline test/e2e/tests/submitDemoRequest.e2e.js]: use `.contains` instead of `. eql` to avoid spaces matching

**Review by AlexanderC — CHANGES_REQUESTED** [review]

File: test/e2e/tests/submitDemoRequest.e2e.js (hunk 4 above)
Comment (avozicov) [reply test/e2e/tests/submitDemoRequest.e2e.js]: I got it, thank you.

Latest status: Fixed

File: test/e2e/tests/assertApiPage.e2e.js (hunk 1 above)
Comment (avozicov) [reply test/e2e/tests/assertApiPage.e2e.js]: I got it, thank you.

Latest status: Fixed

File: test/e2e/tests/assertFooter.e2e.js (hunk 2 above)
Comment (avozicov) [reply test/e2e/tests/assertFooter.e2e.js]: I got it, thank you.

Latest status: Fixed

File: test/e2e/tests/submitDemoRequest.e2e.js (hunk 3 above)
Comment (avozicov) [reply test/e2e/tests/submitDemoRequest.e2e.js]: I got it, thank you.

Latest status: Fixed

**Review by AlexanderC — APPROVED** [review]

**Comment by AlexanderC** [comment]
Please update recink.yml from project root to use the tests and dependencies

**Pull request merged**